### PR TITLE
Enable replacement of millis() function

### PR DESCRIPTION
**Pull request by noseglasses**
For my Kaleidoscope-Python module that wraps Kaleidoscope-Hardware-Virtual I need to be able to replace the `millis()` function that does its own time handling. To achieve this, I turned the function into a weak symbol.

### CODE REVIEW

File: src/VirtualHID/ConsumerControl.h [hunk 1]
```diff
@@ -27,6 +27,7 @@ class ConsumerControl_ {
     void press(uint16_t m);
     void release(uint16_t m);
     void releaseAll(void);
+    void sendReport();
```
Comment (noseglasses) [inline src/VirtualHID/ConsumerControl.h]: This had to be added to adapt to changes in other Kaleidoscope modules.

File: src/VirtualHID/Mouse.cpp [hunk 2]
```diff
@@ -17,12 +18,13 @@ void Mouse_::click(uint8_t b) {
   move(0,0,0);
 }
 
-void Mouse_::move(signed char x, signed char y, signed char wheel) {
+void Mouse_::move(signed char x, signed char y, signed char vWheel, signed char hWheel) {
```
Comment (noseglasses) [inline src/VirtualHID/Mouse.cpp]: Adapted to changes in other Kaleidoscope modules.

File: src/VirtualHID/SingleAbsoluteMouse.cpp [hunk 3]
```diff
@@ -1,20 +1,21 @@
...
 }
 
 // Everything else is stubs for now - no effect
 void SingleAbsoluteMouse_::begin(void) {}
-void SingleAbsoluteMouse_::move(signed char x, signed char y, signed char wheel) {}
-void SingleAbsoluteMouse_::moveTo(uint16_t x, uint16_t y, signed char wheel) {}
-void SingleAbsoluteMouse_::click(uint8_t buttons) {}
-void SingleAbsoluteMouse_::press(uint8_t buttons) {}
-void SingleAbsoluteMouse_::release(uint8_t buttons) {}
+void SingleAbsoluteMouse_::move(signed char /*x*/, signed char /*y*/, signed char /*wheel*/) {}
```
Comment (noseglasses) [inline src/VirtualHID/SingleAbsoluteMouse.cpp]: Caused unused parameter warnings.

**Comment by noseglasses** [comment]
No idea why the travis tests for this project fail. It seems that travis is not correctly configured.

**Comment by cdisselkoen** [comment]
There's a whole lot going on in this diff, and I fail to see how most of it relates to replacing the `millis()` function - would you mind splitting this into multiple PRs by topic?

**Comment by noseglasses** [comment]
I split this bad PR up into two separate PRs. 
Sorry for the bad quality. I am still adapting my work habits to generate more independent commits that are suited for PRs.

**Pull request closed without merging**